### ISSUE TITLE
redux-form: compilation errors with strictFunctionTypes

### ISSUE THREAD

**Issue by jgoz** [DefinitelyTyped/DefinitelyTyped#20414]
- [ ] I tried using the `@types/react-forms` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @carsonf, @aikoven, @LKay, @bancek, @alsiola

As part of #20409, I made a minor change to the `redux-form` tests and found they were failing when ran against TypeScript@next due to `strictFunctionTypes`. I don't use this package and am not familiar with it, but in the interest of getting things passing in that PR, I disabled the option. Note that the errors are present in `master` too.

Example error:

```
file: '.../DefinitelyTyped/types/redux-form/redux-form-tests.tsx'
severity: 'Error'
message: 'Argument of type 'typeof TestFormComponent' is not assignable to parameter of type 'ComponentType<InjectedFormProps<TestFormData, {}>>'.
  Type 'typeof TestFormComponent' is not assignable to type 'StatelessComponent<InjectedFormProps<TestFormData, {}>>'.
    Type 'typeof TestFormComponent' provides no match for the signature '(props: InjectedFormProps<TestFormData, {}> & { children?: ReactNode; }, context?: any): ReactElement<any> | null'.'
at: '68,54'
source: 'ts'
```

There are several such as the above in the main tests and v6 tests.

**Comment by tehbi4** [comment]
There are test and linting fixes for redux-form v7 in #20982.

**Comment by orta** [comment]
Hi thread, we're moving DefinitelyTyped to use [GitHub Discussions](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/53377) for conversations the `@types` modules in DefinitelyTyped.

To help with the transition, we're closing all issues which haven't had activity in the last 6 months, which includes this issue. If you think closing this issue is a mistake, please pop into the [TypeScript Community Discord](https://discord.gg/typescript) and mention the issue in the `definitely-typed` channel.
